src/rules/lowercase-name.ts

```diff
@@ -93,7 +120,7 @@ export default createRule({
         properties: {
           ignore: {
             type: 'array',
-            items: { enum: ['describe', 'test', 'it'] },
+            items: { enum: ['describe', 'test', 'it', 'top'] },
```

Comment:
I think if you're going to do it this way, it would be better to use a boolean option since `top` isn't a `jest` keyword and so doesn't really make sense having it implemented in the same manner of `isTestCase` & `isDescribe`.

That would also be a good position to later extend it to be an array accepting the same properties as `ignore`.

Reply:
My understanding is that if I gave ignore=top, the lowercase rule will not apply to the first parent element(describe/test/it) the inner describe/test should have this rule.
If I do an early return as you suggested the inner elements will also be skipped off the rule. Which is 
```
describe("Xyz abc"){
  test("Cde"){}
}
``` will be a valid one

Reply:
Just to be clear, this is the sort of early return I had in mind:

```
if(ignoreTopLevel && isTopMostJestFunction(node)) {
  return;
}
```

Since `CallExpression` is called for *every* `CallExpression` that ESLint finds, the early return won't interfere with the inner jest function calls.